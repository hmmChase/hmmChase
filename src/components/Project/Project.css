.project {
  width: 80%;
  min-width: 300px;
  border-radius: 10px;
  margin-top: 50px;
  background: #fcf3ca;
}

.project-title {
  margin-left: 20px;
}

.project:last-of-type {
  margin-bottom: 50px;
}

.project-img {
  width: 100%;
  border-top: 1px solid darkgray;
  border-bottom: 1px solid darkgray;
}

.project-description {
  padding: 0 20px;
  margin: 20px 0 0 0;
}

.project-link-container {
  display: flex;
  flex-direction: column;
  padding: 20px;
}

.project-links {
  display: flex;
  justify-content: space-evenly;
}

.project-link-name {
  margin: 0;
  margin-top: 15px;
  font-weight: 600;
}

.project-link {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  border: none;
  margin: 10px 20px 10px 20px;
  color: inherit;
  text-decoration: none;
  cursor: pointer;
  background-color: #c4eada;
  box-shadow: 4px 4px 0 0 #ef7b7b;
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease;
  -o-transition: all 200ms ease;
  transition: all 200ms ease;
}

.project-link:hover {
  transform: translateY(4px) translateX(4px);
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease;
  -o-transition: all 200ms ease;
  transition: all 200ms ease;
  background-color: #ef7b7b;
  box-shadow: -4px -4px 0 0 #c4eada;
}
