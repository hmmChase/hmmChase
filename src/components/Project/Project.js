import React from 'react';
import PropTypes from 'prop-types';
import { ProjectLink } from '@hmmui/hmmchase';
import './Project.css';

const Project = (props) => (
  <article className='project'>
    <h3 className='project-title'>{props.title}</h3>

    <img className='project-img' src={props.img} alt='screenshot' />

    <p className='project-description'>{props.desc}</p>

    <div className='project-link-container'>
      {props.links.map((link, i) => (
        <div key={i}>
          {link.name && <p className='project-link-name'>{link.name}</p>}
          <div className='project-links'>
            <ProjectLink href={link.gitUrl}>GitHub</ProjectLink>

            <ProjectLink href={link.liveUrl}>View Live</ProjectLink>
          </div>
        </div>
      ))}
    </div>
  </article>
);

Project.propTypes = {
  desc: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(
    PropTypes.exact({
      name: PropTypes.string,
      gitUrl: PropTypes.string.isRequired,
      liveUrl: PropTypes.string.isRequired,
    })
  ).isRequired,
  title: PropTypes.string.isRequired,
};

export default React.memo(Project);
