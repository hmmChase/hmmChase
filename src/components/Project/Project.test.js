import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import Project from './Project';

describe('Project', () => {
  it('matches snapshot', () => {
    const mockProps = {
      title: 'mock title',
      desc: 'mock description',
      img: 'mock image url',
      links: [
        {
          name: 'v1. Vanilla JS',
          gitUrl: 'https://github.com/hmmChase/ideabox-v1',
          liveUrl: 'https://hmmchase.github.io/ideabox-v1/',
        },
        {
          name: 'v2. React / RESTful',
          gitUrl: 'https://github.com/hmmChase/ideabox-v2',
          liveUrl: 'https://ideabox-cra-rest.herokuapp.com/',
        },
        {
          name: 'v3. SSR React / GraphQL',
          gitUrl: 'https://github.com/hmmChase/ideabox-v3',
          liveUrl: 'https://ideabox-v3.hmmchase.now.sh/',
        },
        {
          name: 'v4. v3 + user accounts',
          gitUrl: 'https://github.com/hmmChase/ideabox-v4',
          liveUrl: 'https://ideabox-v4.hmmchase.now.sh/',
        },
      ],
    };

    const utils = render(<Project {...mockProps} />);

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
  });
});
