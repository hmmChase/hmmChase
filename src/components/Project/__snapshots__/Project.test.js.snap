// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project matches snapshot 1`] = `
<article
  class="project"
>
  <h3
    class="project-title"
  >
    mock title
  </h3>
  <img
    alt="screenshot"
    class="project-img"
    src="mock image url"
  />
  <p
    class="project-description"
  >
    mock description
  </p>
  <div
    class="project-link-container"
  >
    <div>
      <p
        class="project-link-name"
      >
        v1. Vanilla JS
      </p>
      <div
        class="project-links"
      >
        <a
          class="project-link"
          href="https://github.com/hmmChase/ideabox-v1"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a
          class="project-link"
          href="https://hmmchase.github.io/ideabox-v1/"
          rel="noopener noreferrer"
          target="_blank"
        >
          View Live
        </a>
      </div>
    </div>
    <div>
      <p
        class="project-link-name"
      >
        v2. React / RESTful
      </p>
      <div
        class="project-links"
      >
        <a
          class="project-link"
          href="https://github.com/hmmChase/ideabox-v2"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a
          class="project-link"
          href="https://ideabox-cra-rest.herokuapp.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          View Live
        </a>
      </div>
    </div>
    <div>
      <p
        class="project-link-name"
      >
        v3. SSR React / GraphQL
      </p>
      <div
        class="project-links"
      >
        <a
          class="project-link"
          href="https://github.com/hmmChase/ideabox-v3"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a
          class="project-link"
          href="https://ideabox-v3.hmmchase.now.sh/"
          rel="noopener noreferrer"
          target="_blank"
        >
          View Live
        </a>
      </div>
    </div>
    <div>
      <p
        class="project-link-name"
      >
        v4. v3 + user accounts
      </p>
      <div
        class="project-links"
      >
        <a
          class="project-link"
          href="https://github.com/hmmChase/ideabox-v4"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <a
          class="project-link"
          href="https://ideabox-v4.hmmchase.now.sh/"
          rel="noopener noreferrer"
          target="_blank"
        >
          View Live
        </a>
      </div>
    </div>
  </div>
</article>
`;
